python/gink/impl/key_set.py

```diff
@@ -0,0 +1,222 @@
+            pass
+        Container.__init__(self, muid=muid, database=database)
+        if contents:
+            self.clear(bundler=bundler)
+            # self.update(contents, bundler=bundler)
+        if immediate and len(bundler):
+            self._database.commit(bundler)
+
+    def add(self, key: UserKey, *, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Adds a specified value to the key set """
+        return self._add_entry(key=key, value=inclusion, bundler=bundler, comment=comment)
+    
+    def update(self, keys: Iterable[UserKey], bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Adds multiple specified values to the key set """
+        # Probably best to initialize a bundler here
+        for key in keys:
+            self._add_entry(key=key, value=inclusion, bundler=bundler, comment=comment)
+
+    def contains(self, key: UserKey, as_of: GenericTimestamp=None):
+        """ Returns a boolean stating whether the specified key is in the key set """
+        as_of = self._database.resolve_timestamp(as_of)
+        found = self._database.get_store().get_entry_by_key(self.get_muid(), key=key, as_of=as_of)
+
+        return False if found is None or found.builder.deletion else True
+
+    def discard(self, key: UserKey, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Deletes a specified entry from the key set """
+        return self._add_entry(key=key, value=deletion, bundler=bundler, comment=comment)
+    
+    def remove(self, key: UserKey, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Deletes a specified entry from the key set, but returns KeyError if not found """
+        as_of = self._database.get_now()
+        found = self._database.get_store().get_entry_by_key(self.get_muid(), key=key, as_of=as_of)
+        if found is None or found.builder.deletion:  # type: ignore
+            raise KeyError("Key does not exist")
+        else:
+            return self._add_entry(key=key, value=deletion, bundler=bundler, comment=comment)
+    
+    def pop(self, key: UserKey, bundler: Optional[Bundler]=None, comment: Optional[str]=None, default=None):
+        """ If key exists in the mapping, returns the corresponding value and removes it.
+
+            Otherwise returns default.  In the case that the key is found and removed,
+            then the change is added to the bundler (or committed immedately with comment
+            if no bundler is specified.)
+        """
+        as_of = self._database.get_now()
+        found = self._database.get_store().get_entry_by_key(self.get_muid(), key=key, as_of=as_of)
+        if found is None or found.builder.deletion:  # type: ignore
+            return default
+        self._add_entry(key=key, value=deletion, bundler=bundler, comment=comment)
+        return decode_key(found.builder)
+    
+    def issuperset(self, subset: Iterable, *, as_of: GenericTimestamp=None) -> bool:
```

Comment:
You should specify that subset is specifically of type Iterable[UserKey].
Also, you need to propagate the as_of to the contains method otherwise it doesn't do anything.